Sources/TuistKit/Storyboards/StoryboardGenerator.swift

```diff
@@ -0,0 +1,105 @@
+import Basic
+import Foundation
+import TuistCore
+
+enum StoryboardGenerationError: FatalError, Equatable {
+    case alreadyExisting(AbsolutePath)
+    case launchScreenUnsupported(Platform, Product)
+
+    var description: String {
+        switch self {
+        case let .alreadyExisting(path):
+            return "A storyboard already exists at path \(path.asString)"
+        case let .launchScreenUnsupported(platform, product):
+            return "A \(platform) \(product) does not support a launch screen storyboard"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .alreadyExisting: return .abort
+        case .launchScreenUnsupported: return .abort
+        }
+    }
+
+    static func == (lhs: StoryboardGenerationError, rhs: StoryboardGenerationError) -> Bool {
+        switch (lhs, rhs) {
+        case let (.alreadyExisting(lhsPath), .alreadyExisting(rhsPath)):
+            return lhsPath == rhsPath
+        case let (.launchScreenUnsupported(lhsPlatform), .launchScreenUnsupported(rhsPlatform)):
+            return lhsPlatform == rhsPlatform
+        default:
+            return false
+        }
+    }
+}
+
+protocol StoryboardGenerating: AnyObject {
+    func generate(path: AbsolutePath,
+                  name: String,
+                  platform: Platform,
+                  product: Product,
+                  isLaunchScreen: Bool) throws
+}
+
+final class StoryboardGenerator: StoryboardGenerating {
+    // MARK: - Attributes
+
+    private let fileHandler: FileHandling
+
+    // MARK: - Init
+
+    init(fileHandler: FileHandling = FileHandler()) {
+        self.fileHandler = fileHandler
+    }
+
+    func generate(path: AbsolutePath, name: String, platform: Platform, product: Product, isLaunchScreen: Bool) throws {
+        if isLaunchScreen, product == .app, !platform.supportsLaunchScreen {
+            throw StoryboardGenerationError.launchScreenUnsupported(platform, product)
+        }
+
+        let storyboardPath = path.appending(component: "\(name).storyboard")
+
+        if fileHandler.exists(storyboardPath) {
+            throw StoryboardGenerationError.alreadyExisting(storyboardPath)
+        }
+
+        try StoryboardGenerator.xcstoarybaordContent(platform: platform, isLaunchScreen: isLaunchScreen)
+            .write(to: storyboardPath.url,
+                   atomically: true,
+                   encoding: .utf8)
+    }
+
+    static func xcstoarybaordContent(platform _: Platform, isLaunchScreen: Bool) -> String {
```

Comment:
see the `platform` parameter isn't used, we could omit it

Reply:
I’ve only got this working for iOS at the moment. I just need to confirm that the contents of a storyboard is the same for iOS, macOS & tvOS before I remove this parameter.

If it turns out they can all use the same then I’ll remove it.

Reply:
Yer I've added the support for `macOS` & `tvOS` and they have differences in their storyboards. All the content is from what Apple provide as a default storyboard. The macOS version has a lot of default content.